runtime/src/main/java/org/corfudb/protocols/logprotocol/SMREntryGarbageInfo.java

```diff
@@ -0,0 +1,110 @@
+package org.corfudb.protocols.logprotocol;
+
+
+import io.netty.buffer.ByteBuf;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+import org.corfudb.runtime.CorfuRuntime;
+
+import javax.naming.OperationNotSupportedException;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * SMREntryGarbageInfo maintains garbage information of one SMREntry. This SMREntry could be stand-alone or a part of
+ * MultiSMREntry. The existence of SMREntryGarbageInfo itself indicates the associated SMREntry has been identified
+ * as garbage. SMREntryGarbageInfo could be used individually or composites more complicated ISMRGarbageInfo.
+ */
+@ToString(callSuper = true)
+@NoArgsConstructor
+public class SMREntryGarbageInfo extends LogEntry implements ISMRGarbageInfo {
+
+    /**
+     * Global address of the SMREntry that supersedes the associated SMREntry.
+     */
+    @Getter
+    private long detectionAddress;
+
+    /**
+     * The size of the associated SMREntry in Byte.
+     */
+    private transient int smrEntrySize;
```

Comment:
Remove `transient` and serialize it  and also add size field in SMREntry as we discussed.